### ISSUE TITLE
[npcap]Failed to upgrade npcap when installed in compatible mode.

### ISSUE THREAD

**Issue by yyjdelete** [nmap/nmap#693]
Found that `C:\Windows\SysWOW64\wpcap.dll` and `C:\Windows\System32\wpcap.dll` is not removed by the npcap uninstaller, which make the npcap installer consider that winpcap is already installed.

After delete the 2 files manually, the installer works well.

Tested with 0.80->0.81, Win10 x64 15031(The latest Insyder Preview). And also some old version.

**Comment by Varunram** [comment]
@yyjdelete Seems to work fine for me. Could you specify some steps for reproduction?

**Comment by yyjdelete** [comment]
Unable to reproduct in the same pc now. And I think that thing may only happen after Win10 upgrade between build.

I gen report with `DiagReport.ps1` for 2 pc(not sure whether installed with same param), and found the main difference is cfgs in services (All custom cfgs in `HKLM\SYSTEM\CurrentControlSet\Services\*` was lost, the ones in `HKLM\SOFTWARE\WOW6432Node\Npcap` still exists) and NPF(Not sure what happened with npf)

The one with an fresh npcap 0.81 installed, no wifi.
```ini
*************************************************
Registry Info:
*************************************************
HKLM:\SOFTWARE\WOW6432Node\Npcap:


AdminOnly         : 0
WinPcapCompatible : 1
(default)         : C:\Program Files\Npcap
LoopbackAdapter   : \Device\{D419E1BA-0867-4CA3-A3E6-CADAD3C9912D}



HKLM:\SYSTEM\CurrentControlSet\Services\npcap:


Start              : 1
LoopbackAdapter    : \Device\{D419E1BA-0867-4CA3-A3E6-CADAD3C9912D}
LoopbackSupport    : 1
DltNull            : 1
AdminOnly          : 0
Dot11Support       : 1
VlanSupport        : 1
WinPcapCompatible  : 1
Type               : 1
ErrorControl       : 1
Tag                : 133
ImagePath          : \SystemRoot\system32\DRIVERS\npcap.sys
DisplayName        : @oem37.inf,%NPF_Desc_Standard%;Npcap Packet Driver (NPCAP)
Group              : NDIS
Description        : @oem37.inf,%NPF_Desc_Standard%;Npcap Packet Driver (NPCAP)
NdisMajorVersion   : 6
NdisMinorVersion   : 0
DriverMajorVersion : 1
DriverMinorVersion : 0



HKLM:\SYSTEM\CurrentControlSet\Services\npcap_wifi:


Type               : 1
Start              : 4
ErrorControl       : 1
Tag                : 134
ImagePath          : \SystemRoot\system32\DRIVERS\npcap.sys
DisplayName        : @oem37.inf,%NPF_Desc_WiFi%;Npcap Packet Driver (NPCAP) (Wi-Fi)
Group              : NDIS
Description        : @oem37.inf,%NPF_Desc_WiFi%;Npcap Packet Driver (NPCAP) (Wi-Fi)
NdisMajorVersion   : 6
NdisMinorVersion   : 0
DriverMajorVersion : 1
DriverMinorVersion : 0
                     i



HKLM:\SYSTEM\CurrentControlSet\Services\npf:


Start              : 1
LoopbackAdapter    : \Device\{D419E1BA-0867-4CA3-A3E6-CADAD3C9912D}
LoopbackSupport    : 1
DltNull            : 1
AdminOnly          : 0
Dot11Support       : 1
VlanSupport        : 1
WinPcapCompatible  : 1
Type               : 1
ErrorControl       : 1
Tag                : 131
ImagePath          : \SystemRoot\system32\DRIVERS\npf.sys
DisplayName        : @oem11.inf,%NPF_Desc_Standard%;Npcap Packet Driver (NPF)
Group              : NDIS
Description        : @oem11.inf,%NPF_Desc_Standard%;Npcap Packet Driver (NPF)
NdisMajorVersion   : 6
NdisMinorVersion   : 0
DriverMajorVersion : 1
DriverMinorVersion : 0



HKLM:\SYSTEM\CurrentControlSet\Services\npf_wifi:


Type               : 1
Start              : 4
ErrorControl       : 1
Tag                : 132
ImagePath          : \SystemRoot\system32\DRIVERS\npf.sys
DisplayName        : @oem11.inf,%NPF_Desc_WiFi%;Npcap Packet Driver (NPF) (Wi-Fi)
Group              : NDIS
Description        : @oem11.inf,%NPF_Desc_WiFi%;Npcap Packet Driver (NPF) (Wi-Fi)
NdisMajorVersion   : 6
NdisMinorVersion   : 0
DriverMajorVersion : 1
DriverMinorVersion : 0





*************************************************
Service Info:
*************************************************

Status      : Running
Name        : npcap
DisplayName : Npcap Packet Driver (NPCAP)


Status      : Running
Name        : npf
DisplayName : Npcap Packet Driver (NPF)

```

And the one with npcap 0.80 installed before win10 upgrade, with wifi
```ini
*************************************************
Registry Info:
*************************************************
HKLM:\SOFTWARE\WOW6432Node\Npcap:


AdminOnly         : 0
WinPcapCompatible : 1
(default)         : C:\Program Files\Npcap
LoopbackAdapter   : \Device\{ADC2FDBE-C1CE-430E-A52D-1629515153E4}



HKLM:\SYSTEM\CurrentControlSet\Services\npcap:


Type               : 1
Start              : 3
ErrorControl       : 1
Tag                : 15
ImagePath          : \SystemRoot\system32\DRIVERS\npcap.sys
DisplayName        : @oem0.inf,%NPF_Desc_Standard%;Npcap Packet Driver (NPCAP)
Group              : NDIS
Description        : @oem0.inf,%NPF_Desc_Standard%;Npcap Packet Driver (NPCAP)
NdisMajorVersion   : 6
NdisMinorVersion   : 0
DriverMajorVersion : 1
DriverMinorVersion : 0



HKLM:\SYSTEM\CurrentControlSet\Services\npcap_wifi:


Type         : 1
Start        : 3
ErrorControl : 1
Tag          : 13
ImagePath    : \SystemRoot\system32\DRIVERS\npcap.sys
DisplayName  : @oem0.inf,%NPF_Desc_WiFi%;Npcap Packet Driver (NPCAP) (Wi-Fi)
Group        : NDIS
Description  : @oem0.inf,%NPF_Desc_WiFi%;Npcap Packet Driver (NPCAP) (Wi-Fi)





*************************************************
Service Info:
*************************************************

Status      : Running
Name        : npcap
DisplayName : Npcap Packet Driver (NPCAP)

```

**Comment by dmiller-nmap** [comment]
Npcap had some upgrade problems like this prior to version 0.78. If a process is using Npcap, then the DLLs cannot be deleted and the uninstall will fail. Since 0.78, we detect this and fail the uninstallation. Since 0.78 r3, the uninstaller offers to kill those processes. Please let us know if you are able to reproduce the problem; include information about all screens that show in the uninstaller and installer.

**Comment by yyjdelete** [comment]
@dmiller-nmap 
The same thing happened after update win10 15046 -> 15048, and update npcap 0.8.1->0.8.2.
Update npcap is not required, but update win10 from one build to another build with autoupdate is required.
>All custom cfgs in `HKLM\SYSTEM\CurrentControlSet\Services\*` was removed by windows . And seems npcap don't use the cfg in `HKLM\SOFTWARE\WOW6432Node\Npcap`, so the uninstaller don't know what to do.
>NPF Service and Loopback Adapter is lost


Log is attached.(Forget to copy the log in uninstaller, seems it's successed)
[log.tar.gz](https://github.com/nmap/nmap/files/819157/log.tar.gz)
The installer option is choose all except `Restrict Npcap driver's access to Administrators only`

**Comment by dmiller-nmap** [comment]
This looks like something wrong with the Windows upgrade process. [Citrix XenApp and XenDesktop seem to have the same problem](https://support.citrix.com/article/CTX216312), for which they note, "There is no solution. Citrix is working with Microsoft to resolve this issue." So unlikely we could stop it from happening.

Instead, maybe we can enable the uninstaller to recover from this corrupt state. @yyjdelete do you have any problems reinstalling after the Windows upgrade? Does reinstalling give you a working Npcap?

**Comment by yyjdelete** [comment]
1. Uninstall old Npcap
2. Manually delete `%SystemRoot%\System32\wpcap.dll` and the same thing in `SysWOW64`
3. Reinstall Npcap

This works.

**Comment by dmiller-nmap** [comment]
I believe this was reported and fixed as #906. The first version to have a fix was 0.93, but Npcap 0.95 is available now. Please test [Nmap 0.95](https://nmap.org/npcap/#download) and let us know if it solves this problem.

**Comment by yyjdelete** [comment]
0.94 works well.
